src/lib.rs

```diff
@@ -90,6 +117,15 @@ impl Heap {
         }
     }
 
+    /// Crates a new heap from a slice of raw memory.
+    pub fn with_memory(mem: &'static mut [MaybeUninit<u8>]) -> Heap {
```

Comment:
I'm not sure about the function name. Given the similarity to `init_from_slice`, we could call this function e.g. `new_from_slice` or just `from_slice`.

We should also reference the `init_from_slice` function in the docs since it provides more detailed information.